### DISCUSSION TITLE
Hombrew installation is hung on Monterey

### DISCUSSION THREAD

**Discussion by OmRam** [Homebrew/discussions#2933] [Getting started] [answered]
I am trying to install homebrew on Mac OS Monterey and it is hung at /usr/bin/sudo /usr/bin/touch /tmp/.com.apple.dt.CommandLineTools.installondemand.in-progress.

Any idea what is wrong ?

ram@ram-mac ~ % /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
==> Checking for `sudo` access (which may request your password)...
Password:
==> This script will install:
/opt/homebrew/bin/brew
/opt/homebrew/share/doc/homebrew
/opt/homebrew/share/man/man1/brew.1
/opt/homebrew/share/zsh/site-functions/_brew
/opt/homebrew/etc/bash_completion.d/brew
/opt/homebrew
==> The Xcode Command Line Tools will be installed.

Press RETURN to continue or any other key to abort:
==> /usr/bin/sudo /usr/sbin/chown -R ram:admin /opt/homebrew
==> Searching online for the Command Line Tools
==> /usr/bin/sudo /usr/bin/touch /tmp/.com.apple.dt.CommandLineTools.installondemand.in-progress
^C
ram@ram-mac ~ % /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
==> Checking for `sudo` access (which may request your password)...
Password:
==> This script will install:
/opt/homebrew/bin/brew
/opt/homebrew/share/doc/homebrew
/opt/homebrew/share/man/man1/brew.1
/opt/homebrew/share/zsh/site-functions/_brew
/opt/homebrew/etc/bash_completion.d/brew
/opt/homebrew
==> The Xcode Command Line Tools will be installed.

Press RETURN to continue or any other key to abort:
==> /usr/bin/sudo /usr/sbin/chown -R ram:admin /opt/homebrew
==> Searching online for the Command Line Tools
==> /usr/bin/sudo /usr/bin/touch /tmp/.com.apple.dt.CommandLineTools.installondemand.in-progress

**Comment by gromgit** [comment, marked as answer]
At this stage, the installer is running Apple's own `softwareupdate` utility to check for the latest CLT version. There's nothing to do but wait, especially if you don't have a fast Internet connection.

**Reply by OmRam** [reply]
There is no progress shown either...Its just stuck at this line
==> /usr/bin/sudo /usr/bin/touch /tmp/.com.apple.dt.CommandLineTools.installondemand.in-progress

I am getting 75Mbps for download.

**Reply by gromgit** [reply]
You could tell bash to run the installer in trace mode (print every command before running it), but it'll spew a TON of output:
```
/bin/bash -xc "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
```

**Reply by OmRam** [reply]
This is what I get with trace mode. Just copying the last few lines of the output till the point it gets stuck. Looks like it is running softwareupdate but not sure why it is stuck.
```
==> Searching online for the Command Line Tools
+ clt_placeholder=/tmp/.com.apple.dt.CommandLineTools.installondemand.in-progress
+ execute_sudo /usr/bin/touch /tmp/.com.apple.dt.CommandLineTools.installondemand.in-progress
+ args=("$@")
+ local -a args
+ have_sudo_access
+ [[ ! -x /usr/bin/sudo ]]
+ SUDO=("/usr/bin/sudo")
+ local -a SUDO
+ [[ -n '' ]]
+ [[ -n '' ]]
+ [[ -z 0 ]]
+ [[ -z '' ]]
+ [[ 0 -ne 0 ]]
+ return 0
+ [[ -n '' ]]
+ ohai /usr/bin/sudo /usr/bin/touch /tmp/.com.apple.dt.CommandLineTools.installondemand.in-progress
++ shell_join /usr/bin/sudo /usr/bin/touch /tmp/.com.apple.dt.CommandLineTools.installondemand.in-progress
++ local arg
++ printf %s /usr/bin/sudo
++ shift
++ for arg in '"$@"'
++ printf ' '
++ printf %s /usr/bin/touch
++ for arg in '"$@"'
++ printf ' '
++ printf %s /tmp/.com.apple.dt.CommandLineTools.installondemand.in-progress
+ printf '==> %s\n' '/usr/bin/sudo /usr/bin/touch /tmp/.com.apple.dt.CommandLineTools.installondemand.in-progress'
==> /usr/bin/sudo /usr/bin/touch /tmp/.com.apple.dt.CommandLineTools.installondemand.in-progress
+ execute /usr/bin/sudo /usr/bin/touch /tmp/.com.apple.dt.CommandLineTools.installondemand.in-progress
+ /usr/bin/sudo /usr/bin/touch /tmp/.com.apple.dt.CommandLineTools.installondemand.in-progress
+ clt_label_command='/usr/sbin/softwareupdate -l |
                      grep -B 1 -E '\''Command Line Tools'\'' |
                      awk -F'\''*'\'' '\''/^ *\*/ {print $2}'\'' |
                      sed -e '\''s/^ *Label: //'\'' -e '\''s/^ *//'\'' |
                      sort -V |
                      tail -n1'
+++ /bin/bash -c '/usr/sbin/softwareupdate -l |
                      grep -B 1 -E '\''Command Line Tools'\'' |
                      awk -F'\''*'\'' '\''/^ *\*/ {print $2}'\'' |
                      sed -e '\''s/^ *Label: //'\'' -e '\''s/^ *//'\'' |
                      sort -V |
                      tail -n1'
```

**Reply by gromgit** [reply]
Yup, what I said earlier:
> At this stage, the installer is running Apple's own `softwareupdate` utility to check for the latest CLT version.

75Mbps download is irrelevant when you're waiting on Apple's own servers.

**Reply by carlocab** [reply]
You can try running `softwareupdate -l` yourself to see how long it takes.

**Reply by OmRam** [reply]
can we bypass the `softwareupdate` step?

**Reply by SMillerDev** [reply]
No, it's installing requirements there

**Reply by OmRam** [reply]
Thanks all for the help. Cyberreason was blocking `softwareupdate`. Restarted mac in safe mode and installed it.
